cartography/intel/aws/sagemaker/model_packages.py

```diff
@@ -0,0 +1,161 @@
+import logging
+from typing import Any
+from typing import Dict
+from typing import List
+
+import boto3
+import neo4j
+
+from cartography.client.core.tx import load
+from cartography.graph.job import GraphJob
+from cartography.intel.aws.sagemaker.util import extract_bucket_name_from_s3_uri
+from cartography.models.aws.sagemaker.model_package import (
+    AWSSageMakerModelPackageSchema,
+)
+from cartography.util import aws_handle_regions
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+@timeit
+@aws_handle_regions
+def get_model_packages(
+    boto3_session: boto3.session.Session,
+    region: str,
+) -> List[Dict[str, Any]]:
+    """
+    Get all SageMaker Model Packages in the given region.
+    """
+    client = boto3_session.client("sagemaker", region_name=region)
+    paginator = client.get_paginator("list_model_packages")
+    model_packages: List[Dict[str, Any]] = []
+
+    # Get all model package ARNs
+    model_package_arns: List[str] = []
+    for page in paginator.paginate():
+        for package in page.get("ModelPackageSummaryList", []):
+            model_package_arns.append(package["ModelPackageArn"])
+
+    # Get detailed information for each model package
+    for package_arn in model_package_arns:
+        try:
```

Comment:
I don't think we need to have a `try+except` here because aws_handle_regions takes care of it.